janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/cache/CacheTransaction.java

```diff
@@ -129,9 +134,7 @@ private KCVMutation convert(KCVEntryMutation mutation) {
     private void flushInternal() throws BackendException {
         if (numMutations > 0) {
             //Consolidate all mutations prior to persistence to ensure that no addition accidentally gets swallowed by a delete
-            for (Map<StaticBuffer, KCVEntryMutation> store : mutations.values()) {
-                for (KCVEntryMutation mut : store.values()) mut.consolidate();
-            }
+            mutations.values().parallelStream().forEach(store -> store.values().forEach(KCVEntryMutation::consolidate));
```

Comment:
I have no objection but have you considered the negative implications to small transactions?

Reply:
Perhaps we could introduce some threshold limit. In case the mutations amount is less then `N` then we kick the old sequential approach. Otherwise, we use parallel approach. I guess this `N` could be different on different servers, but if we add a configuration parameter to that then users could configure their own `N`. 
As the default threshold limit we could use something that is good enough on different servers.
I didn't test it myself, so also interesting to know when one approach is better than another.

Reply:
Done.

```
    public static final ConfigOption<Integer> NUM_MUTATIONS_PARALLEL_THRESHOLD = new ConfigOption<>(STORAGE_NS,"num-mutations-parallel-threshold",
        "Number of mutations after which use parallel processing for transaction aggregations.",
    ConfigOption.Type.MASKABLE, 100, ConfigOption.positiveInt());
```